src/Serializer/AbstractItemNormalizer.php

```diff
@@ -161,7 +162,7 @@ protected function getAllowedAttributes($classOrObject, array $context, $attribu
     protected function setAttributeValue($object, $attribute, $value, $format = null, array $context = [])
     {
         if (!\is_string($attribute)) {
-            throw new InvalidArgumentException('Invalid value provided (invalid IRI?).');
+            throw new InvalidIriException();
```

Comment:
```
 throw new InvalidIriException('Invalid value provided (invalid IRI?).');
```

Reply:
What would be the point of the exception if the message is not always the same one ? It's like using a generic exception... look at the exceptions from symfony, they each have somehting customisable but a defined (and thus customizable) message (even though the message is also quite pointless and not used)

Reply:
Because we can have different reasons for an invalid IRI exception:
- bad format
- route not found
- data not found

for example

Reply:
Yeah that's why I had the `(invalid IRI?)` variable, so we would always have `Invalid value provider (REASON)`. But `¯\_(ツ)_/¯` because it's the same thing actually. :}

Reply:
I agree with @soyuka (and every other exception we have for now work like this). You can also add the message as the default value of the `$message` parameter.

Reply:
Yep, changed and removed the constructor altogether, as @soyuka suggested https://github.com/api-platform/core/pull/1855#discussion_r181410250